src/net/http.rs

```diff
@@ -2516,46 +2564,55 @@ impl HttpRequestType {
                 _key,
                 tip_opt,
                 with_proof,
+                use_latest_tip,
```

Comment:
You don't have to address it here, but could you open an issue to consolidate `tip_opt`, `with_proof`, and `use_latest_tip` into a single struct that contains the standard query parameters that most API endpoints accept?  Otherwise PRs like this are going to get untenable ;)

Reply:
sure!